### PR TITLE
fix(theme): waits for route to complete before changing out theme

### PR DESCRIPTION
**Pull request by dzucconi**
@damassi — thanks for pointing me in the right direction with this one.

### CODE REVIEW

File: src/v2/Apps/Components/AppShell.tsx [hunk 1]
```diff
@@ -6,11 +6,12 @@ import { NAV_BAR_HEIGHT, NavBar, MOBILE_NAV_HEIGHT } from "v2/Components/NavBar"
 import { Match } from "found"
 import { isFunction } from "lodash"
 import { Footer } from "v2/Components/Footer"
-import React, { useEffect } from "react"
+import React, { useEffect, useState } from "react"
 import createLogger from "v2/Utils/logger"
 import { useSystemContext } from "v2/Artsy"
 import { HorizontalPadding } from "v2/Apps/Components/HorizontalPadding"
 import { AppContainer } from "./AppContainer"
+import { useRouteComplete } from "v2/Utils/Hooks/useRouteComplete"
```
Comment (damassi) [inline src/v2/Apps/Components/AppShell.tsx:14]: 👍

File: src/v2/Apps/Components/AppShell.tsx [hunk 2]
```diff
@@ -48,6 +49,11 @@ export const AppShell: React.FC<AppShellProps> = props => {
     document.body.setAttribute("data-test", "AppReady")
   }, [])
 
+  // Wait for route to finish rendering before (possibly) switching out the theme
```
Comment (damassi) [inline src/v2/Apps/Components/AppShell.tsx:52]: This is small but could we expand this comment a bit and just kinda explain whats going on here and why its needed?

**Review by damassi — APPROVED** [review]
One tiny req, otherwise looks great 👍 I like that we can now configure things through our routes file too.

File: src/v2/Apps/Components/AppShell.tsx [hunk 3]
```diff
@@ -48,6 +49,17 @@ export const AppShell: React.FC<AppShellProps> = props => {
...
 
+  /**
+   * Wait for route to finish rendering before (possibly) switching out the theme.
+   *
+   * When the route changes, the configured theme will change immediately; this
+   * will cause the styles to update out of sync with the page change. Here we
+   * wait for the route to finish rendering before setting the next theme.
+   */
+  const nextTheme = routeConfig.theme ?? "v2"
+  const [theme, setTheme] = useState<"v2" | "v3">(nextTheme)
```
Comment (damassi) [inline src/v2/Apps/Components/AppShell.tsx:60]: @dzucconi - Oh sorry, i was thinking more about mentioning how we're migrating pages and how some have old palette and new and that's why this is needed. 

(feel free to update or not, its not a blocker or that big of a deal, i can just see it being somewhat confusing down the line)

Comment (dzucconi) [reply src/v2/Apps/Components/AppShell.tsx:60]: 😄  Will sneak in an update in my next PR

**Pull request merged**